ios/FluentUI/Core/Colors.swift

```diff
@@ -281,67 +281,150 @@ public final class Colors: NSObject {
 
     /// Use these funcs to grab a color customized by a ColorProviding object for a specific window.. If no colorProvider exists for the window, falls back to deprecated singleton theme color
     @objc public static func primary(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryColor(for: window) ?? primary
+        return colorProvidersMap.object(forKey: window)?.primaryColor(for: window) ?? FallbackThemeColor.primary
     }
 
     @objc public static func primaryTint10(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryTint10Color(for: window) ?? primaryTint10
+        return colorProvidersMap.object(forKey: window)?.primaryTint10Color(for: window) ?? FallbackThemeColor.primaryTint10
     }
 
     @objc public static func primaryTint20(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryTint20Color(for: window) ?? primaryTint20
+        return colorProvidersMap.object(forKey: window)?.primaryTint20Color(for: window) ?? FallbackThemeColor.primaryTint20
     }
 
     @objc public static func primaryTint30(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryTint30Color(for: window) ?? primaryTint30
+        return colorProvidersMap.object(forKey: window)?.primaryTint30Color(for: window) ?? FallbackThemeColor.primaryTint30
     }
 
     @objc public static func primaryTint40(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryTint40Color(for: window) ?? primaryTint40
+        return colorProvidersMap.object(forKey: window)?.primaryTint40Color(for: window) ?? FallbackThemeColor.primaryTint40
     }
 
     @objc public static func primaryShade10(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryShade10Color(for: window) ?? primaryShade10
+        return colorProvidersMap.object(forKey: window)?.primaryShade10Color(for: window) ?? FallbackThemeColor.primaryShade10
     }
 
     @objc public static func primaryShade20(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryShade20Color(for: window) ?? primaryShade20
+        return colorProvidersMap.object(forKey: window)?.primaryShade20Color(for: window) ?? FallbackThemeColor.primaryShade20
     }
 
     @objc public static func primaryShade30(for window: UIWindow) -> UIColor {
-        return colorProvidersMap.object(forKey: window)?.primaryShade30Color(for: window) ?? primaryShade30
+        return colorProvidersMap.object(forKey: window)?.primaryShade30Color(for: window) ?? FallbackThemeColor.primaryShade30
     }
 
     private static var colorProvidersMap = NSMapTable<UIWindow, ColorProviding>(keyOptions: .weakMemory, valueOptions: .weakMemory)
 
+    fileprivate enum FallbackThemeColor {
```

Comment:
Avoid using fileprivate. If you have Swiftlint installed, it should be giving you a warning for that.

Private over fileprivate Violation: Prefer `private` over `fileprivate` declarations. (private_over_fileprivate)

Reply:
If I change `fileprivate` to `private` there will be compiler error like "'primaryShade30' is inaccessible due to 'private' protection level". Should we add special comment to silence Swiftlint warnings, or should we change `private` into `internal` instead (though no other parts of FluentUI should access these values)?

Reply:
is it because you didn't remove the "private" on the variable itself?

Reply:
Yes! Preserve `private` only in type declaration solved this problem.